scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -169,6 +174,101 @@ def _value_or_config(data, key, config):
     return data[key] if key in data else config.get(key)
 
 
+class EnumType:
+    def __init__(self, enum: Enum):
+        self.type = enum.name
+        self.base_type = enum.base_type.lower()
```

Comment:
Why are we doing case-folding here?  At the very least this needs to be documented.

Reply:
Removed. I double checked that it does not changed anything.